### PR TITLE
Assorted updates for GEOS-Chem 1-year benchmarking -- closes #163

### LINKED ISSUE #163: [BUG/ISSUE] Inconsistent restart file path usage in 1-year benchmarks
There is a discrepancy in the way GCPy 1-year benchmark code uses the restart file directories set in the 1-year benchmark yml files. Some benchmark code, such as the code for mass tables, gets the restart file paths using function `get_filepaths`, which expects the path to the directory in which the `Restarts` folder lives. The returned path has `Restarts` appended to the input directory, along with the filename. In contrast, the Ox budget code expects an input restart file path to include `Restarts` and does not rely on `get_filepaths` to retrieve the restart file path and name.

This is a problem because the restart file path set in the yml file wiill either break the mass table code if `Restarts` is included, or will break the Ox budget code if it is not.

### PR DESCRIPTION
**Pull request by lizziel**
This update includes the implementation of diff-of-diff plots for 1-year GCHP benchmarks, among other things. Here is a list of the updates:

1. Use lat/lon 1x1.25 for GCHP vs GCHP comparisons in 1-year benchmark. This avoids an issue with cubed-sphere to cubed-sphere regridding that is encountered in the 14.0 1-yr GCHP benchmark due to different grid resolution from the last 1-yr GCHP benchmark (c24 vs c48).
2. Regrid GCHP ref and GCHP dev to C48 using [sparselt](https://github.com/liambindle/sparselt) package prior to taking the difference for 1-yr benchmark diff-of-diff plots. This is a simpler way to regrid than using the esmf python tools implemented in GCPy. Previously no regridding was done prior to computing differences for diff-of-diffs since GCHP ref and dev were the same grid resolution. This is not the case for the GCHP 1-yr benchmark of 14.0. It will be the case, however, moving forward, and this code will no longer be needed. 
3. Fix bugs in 1-yr GCHP benchmark code that have to do with restart file paths
4. Set a single benchmark results directory for 1-yr benchmarks and put all benchmark artifacts in that folder, including both GCHP comparisons and GCC vs GCC comparisons. Previously the GCC vs GCC comparison were written to the GCC dev folder and all GCHP comparisons were written to the GCHP dev folder. This was problematic when running the benchmark on other's people data (permissions issue writing to their rundir) or when using data not stored in a run directory.
5. Append load to calls to open data to avoid dask arrays when reading via open_mfdataset. Dask arrays cause a problem in sparselt.
6. Comment out using fake dimension for GCHP vs GCC diff-of-diffs plots. This code was added somewhat recently for the case of ref and dev not having the same time. However, it causes problems when there are more than one time in the file. The quick fix is to disable this feature.
7. Add the implementation of diff-of-diffs to 1-year benchmark code. Previously it was only implemented for 1-month benchmarks. The 1-yr benchmarks are different in that they have 12 months of data (time series length 12), rather than a single time in the files.

### CODE REVIEW

**Comment by lizziel** [comment]
I am marking this PR for review. @yantosca is concurrently working on benchmark code in GCPy for the upcoming GEOS-Chem 14.0 benchmark, and may still push to this branch. I am marking him as reviewer so that this does not get merged until he gives the ok.

**Comment by yantosca** [comment]
Also added the following updates:
  8. Removed hardwired restart-file paths from `util.get_filepath` and `util.get_filepaths`.  We now specify the subdirectories of Ref and Dev with `outputs_subdir` and `restarts_subdir` tags in the benchmark YAML files.
  9. Make sure the verbose keyword is used consistently in the benchmark routines
  10. Removed references the `gcc_is_pre_14_0` tag from the benchmark YAML files and benchmark and files `run_benchmark.py` and `util.py`.  We now specify the path to the restart files, so this is now obsolete.
  11. Removed multiple environment files; we now only keep one environment file: `docs/environment_files/environment.yml`.  (I confirmed that we could build conda environments with these).  Python package versions were also updated (esp. Sphinx and its dependencies).
  12. Added `docs/requirements.yml` with explicit package versions that are known to work. This is used when documentation is built at `gcpy.readthedocs.io` -- the `requirements.txt` file is used to initialize a PyPi environment.
  13. Add updates to restore GCHP vs GCC and GCHP vs GCC mass tables and Ox budget functionality.

**Comment by yantosca** [comment]
Also added:
  14. Fixed several issues tha tprevented 1-year TransportTracers plots and tables from being generated.

**Comment by yantosca** [comment]
Also added:
15. Allow specification of species database directory

**Review by yantosca — COMMENTED** [review]
Some more updates are needed.

**Comment by yantosca** [comment]
Also note the changelog updates:

## Unreleased
### Added
- New features in benchmarking scripts (@lizziel, @yantosca)
  - Extra print statements (@lizziel)
  - Diff-of-diffs plots for 1-year benchmarks (@lizziel)
  - sparselt is now a GCPy requirement (@lizziel)
  - Add switch for 
- Removed obsolete environment.yml files (@yantosca)
- Added requirements.yml to docs folder for Sphinx/RTD documentation (@yantosca)
- New regridding script `regrid_restart_file.py` (@liambindle)
### Changed
- Fixed several issues in benchmarking scripts (@laestrada, @lizziel, @yantosca)
  - Add OMP_NUM_THREADS and OMP_STACKSIZE in `plot_driver.sh` (@yantosca)
  - Increase requested memory to 50MB in `plot_driver.sh` (@yantosca)
  - Benchmark scripts print a message upon completion (@yantosca)
  - Linted several benchmarking routines with Pylint (@yantosca)
  - Rewrote algorithm of add_lumped_species_to_dataset for speed (@yantosca)
  - Can now specify the path to species_database.yml for 1yr benchmarks (@yantosca)
  - 1-yr benchmarks now save output in subdirs of the same path (@lizziel)
  - Avoid hardwiring restart file paths in benchmark scripts (@yantosca)
  - Now use outputs_subdir tag from YAML file for paths to diagnostic files (@yantosca)
  - Now use restarts_subdir tag from YAML file for paths to restart files (@yantosca)
  - GCPy now uses proper year for dev in 1-yr benchmarks (@laestrada)
  - Fixed date string issue in benchmarking scripts (@lizziel)
  - Updates for new GCHP restart file format (@lizziel)  
- Updated environment.yml with package versions that work together (@yantosca)
- Updated the AUTHORS.txt and LICENSE.txt files (@yantosca)

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,32 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 ### Added
```
Comment (yantosca) [inline CHANGELOG.md:8]: Updated the changelog for 1.3.0

**Pull request merged**